resource/simplestore.go

```diff
@@ -5,33 +5,23 @@ import (
 	"fmt"
 	"reflect"
 	"time"
-)
 
-// SimpleStoreResource is a type used by SimpleStore to return concrete data, rather than an Object.
-// All data from an underlying Object returned by a client is preserved in this struct,
-// But with a concrete type for the SpecObject() (presented here as Spec).
-// TODO: concrete ObjectMetadata type in addition to SpecType? Would make using this a mite more complicated with MetadataOption
-type SimpleStoreResource[SpecType any] struct {
-	Spec           SpecType       `json:"spec"`
-	StaticMetadata StaticMetadata `json:"staticMetadata"`
-	CommonMetadata CommonMetadata `json:"objectMetadata"`
-	CustomMetadata CustomMetadata `json:"customMetadata"`
-	Subresources   map[string]any `json:"subresources"`
-}
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
 
 // ObjectMetadataOption is a function which updates an ObjectMetadata
-type ObjectMetadataOption func(o *CommonMetadata)
+type ObjectMetadataOption func(o *metav1.ObjectMeta)
```

Comment:
Should we create aliases for these types to avoid those imports in end-user code?